paddle/fluid/operators/listen_and_serv_op.cc

```diff
@@ -258,6 +313,7 @@ void ListenAndServOp::RunImpl(const framework::Scope &scope,
 
   bool sync_mode = Attr<bool>("sync_mode");
   auto fan_in = Attr<int>("Fanin");
+  auto inputs = Inputs("X");
```

Comment:
Some variable may not in the inputs, need to check.

Reply:
Seems all the received gradients should be in the input arguments:

https://github.com/PaddlePaddle/Paddle/blob/a39eba77eb44f7b56159b5ae71bb37760f955a83/python/paddle/fluid/transpiler/distribute_transpiler.py#L497-L500

Reply:
The gradients of dist lookup table would also been added into the received list:

https://github.com/PaddlePaddle/Paddle/blob/2fd1bf2ea6b6e0d27fb49461dd2b35c8e2a2b13b/python/paddle/fluid/transpiler/distribute_transpiler.py#L890-L891